lib/charms/layer/azure.py

```diff
@@ -139,15 +139,16 @@ def send_additional_metadata(request):
     Get additional info about the requesting instance via the API that isn't
     available from the metadata server.
     """
+    run_config = hookenv.config() or {}
     res_grp = _azure('group', 'show', '--name', request.resource_group)
     # hard-code most of these because with Juju, they're always the same
     # and the queries required to look them up are a PITA
     request.send_additional_metadata(
         resource_group_location=res_grp['location'],
-        vnet_name='juju-internal-network',
-        vnet_resource_group=request.resource_group,
-        subnet_name='juju-internal-subnet',
-        security_group_name='juju-internal-nsg',
+        vnet_name=run_config.get('vnetName') if run_config.get('vnetName') else 'juju-internal-network',
+        vnet_resource_group=run_config.get('vnetResourceGroup') if run_config.get('vnetResourceGroup') else request.resource_group,
```

Comment:
It looks like this will change the default value of vnet_resource_group to be `'juju-default-rg'` (since that's the charm config's default) instead of `request.resource_group`.

Perhaps the vnetResourceGroup charm config's default value should be `''` to preserve the default behavior where `request.resource_group` is used instead?

Reply:
Hi Cynerva, yes that is the idea. 
If I have my VMs on RG A and network on RG B, I need to override the vnet_resource_group to aim at RG B for vnet.
We should be able to override this config if I want, otherwise use what is defined on request's resource group. Does that make sense?

Reply:
Sorry, let me rewrite here, you are right, let me move it to empty